### ISSUE TITLE
JIT inputs does not support collections.OrderedDict

### ISSUE THREAD

**Issue by Max-ChenFei** [pytorch/pytorch#23258]
## 🐛 Bug

I choose `model = torchvision.models.segmentation.deeplabv3_resnet101(pretrained=True)` and then use `torch.jit.get_trace_graph(model,  inputs)` to get a graph, a `RuntimeError: Only tuples, lists, and Variables supported as JIT inputs, but got collections.OrderedDict` is in `torch\jit\__init__.py` line 295. 

I think if the outputs of the model are the collections.OrderedDict, just take the **values of the dict** and then convert to **tuple**.
## To Reproduce

1. model = torchvision.models.segmentation.deeplabv3_resnet101(pretrained=True)
2. model.eval()
3. example = torch.rand(1, 3, 224, 224)
4. torch.jit.get_trace_graph(model, example)

## Expected behavior

The graph could get for the `deeplabv3_resnet101`.

## Environment

 - PyTorch Version (e.g., 1.0): 1.1.0
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6.8
 - CUDA/cuDNN version: 9.0
 - GPU models and configuration:
 - Any other relevant information: torchvision: 0.3.0

**Comment by zdevito** [comment]
Can you try this with `torch.jit.trace`? `get_trace_graph` is not the public API and we should do a better job of hiding it. If the problem persists, you can simply put the Module into another Module that extracts the results that are desired out of the dictionary and returns them.

**Comment by Max-ChenFei** [comment]
Thanks for your reply. 
The problem still persists. Putting a module to another module is a simple way,
but this Module is from **torchvision.models.segmentation**. 
Will you support all the module from **torchvision**?

**Comment by lara-hdr** [comment]
Hello @Max-ChenFei, 
I think the issue is in the outputs rather than the inputs (even if the error says it's in the inputs).
A fast fix would be to change the output of _SimpleSegmentationModel to be a list instead of a dict (I was able to run it with "return list(result.values())" instead of "return result").

**Comment by tribbloid** [comment]
Also affects me, except that my model is:

```
from torchvision.models.detection.backbone_utils import resnet_fpn_backbone
fpn = resnet_fpn_backbone("resnet18", True)
```
the following code will trigger a similar error:

```
device = torch.device('cpu')
x = torch.rand(1, 3, 300, 300, dtype=torch.float32, device=device)
traced = torch.jit.trace(fpn, x)

  File ".../spike_fpn.py", line 36, in <module>
    traced = torch.jit.trace(fpn, x)
  File "/home/shared/venv/dl-torch/lib/python3.7/site-packages/torch/jit/__init__.py", line 772, in trace
    check_tolerance, _force_outplace, _module_class)
  File "/home/shared/venv/dl-torch/lib/python3.7/site-packages/torch/jit/__init__.py", line 904, in trace_module
    module._c._create_method_from_trace(method_name, func, example_inputs, var_lookup_fn, _force_outplace)
RuntimeError: Tracer cannot infer type of OrderedDict([(0, tensor([[[[-2.4962e-01, -4.3837e-01, -2.9789e-01,  ..., -1.7870e-01,
```

@lara-hdr Is it very hard to support dict/orderedDict/namedTuple in the output or intermediate representation?

**Comment by zhuj06** [comment]
I get the exactly same problem when I trying to convert this model to onnx version. It also call the trace.

**Comment by lara-hdr** [comment]
This error should be fixed in PyTorch 1.3 for exporting to ONNX, with https://github.com/pytorch/pytorch/pull/25889

**Comment by zhuj06** [comment]
@lara-hdr Thanks, it worked.

**Comment by eellison** [comment]
Closing bc issue was fixed with #25889

**Comment by ikamensh** [comment]
Named tuples don't work as of now:

```python
from collections import namedtuple

import torch
print(torch.version.__version__)  # 1.9.0; same for 1.7.0 and 1.8.1

A = namedtuple("A", "x y")

def foo(ntpl : A):
    return ntpl.x


a = A(torch.zeros([1,]), torch.ones([1,]) )

print(foo(a))
traced = torch.jit.trace(foo, (a, ))  # AttributeError: 'tuple' object has no attribute 'x'
```
